File: backend-spring-boot/src/main/java/com/placeti/avaliacao/model/Comercio.java
package com.placeti.avaliacao.model;

import com.placeti.avaliacao.Enum.TipoComercio;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "comercio")
public class Comercio {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID", nullable = false)    
    private Long id;

    @Column(name = "NOME", nullable = false)    
    private String nome;

    @Column(name = "RESPONSAVEL", nullable = false)    
    private String responsavel;

    @Column(name = "TIPO", nullable = false)    
    @Enumerated(EnumType.STRING)
    private TipoComercio tipo;

    @ManyToOne 
    @JoinColumn(name = "cidade_id", nullable = false)
    private Cidade cidade;
}

File: frontend-angular/src/domain/comercio.ts


export class Comercio {
    id?: number;
    nome: string = '';
    responsavel: string = '';
    tipo: TipoComercio;
    cidadeId: number;
  
    constructor(nome?: string, responsavel?: string, tipo?: TipoComercio, cidadeId?: number) {
      this.nome = nome || '';
      this.responsavel = responsavel || '';
      this.tipo = tipo!;
      this.cidadeId = cidadeId!;
    }
  }
  
  export enum TipoComercio {
    FARMACIA = 'FARMACIA',
    PADARIA = 'PADARIA',
    POSTO_GASOLINA = 'POSTO_GASOLINA',
    LANCHONETE = 'LANCHONETE'
    
  }
  

